src/data/adaptive_context.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "common/file_stream.h"
+#include "data/iterator_facade.h"
+
+namespace marian {
+namespace data {
+
+
+class AdaptiveContextReader;
+
+
+/**
+ * @brief An iterator for easier access of the context sentences produced by
+ * `AdaptiveContextReader::getSamples()`
```

Comment:
Nit: if I remember correctly, we have decided not using `@brief` and enabled a feature, which converts the first sentence (ended with a full stop) as briefing automatically. Please consider changing this throughout the file. Thank you for adding some code documentation.